### PR TITLE
[ready for review] Update world builder to verion 0.4.0.

### PR DESCRIPTION
**Pull request by MFraters**
An update to version 0.4.0-pre in preparation for the new aspect release. Version 0.4.0 of the world builder has not been release yet, but this is to test whether it works or if any changes need to be made before the release. 

Let me know if there are any issues so that they can be fixed before the next world builder release.

### CODE REVIEW

**Comment by gassmoeller** [comment]
The tests seem to pass, is there something else you want us to test? Let us know when the world builder release is done so we can include it. Alternatively: To not change things significantly right before the release we could also update world builder after the release? I do not remember how we handled this in the past.

**Comment by tjhei** [comment]
Yeah, either we update very soon (in a few days) or we wait for after. I don't have a strong preference.

**Comment by MFraters** [comment]
> I do not remember how we handled this in the past.
In the last few years we have always updated the world builder before the release. I would also like to do that in this case since there are some important fixes in the world builder I would like to get in aspect. 

The world builder is as good as ready to be released (this can be seen as a release candiate). The only thing I want to change before it is released is that I have to update the copyright years and just do some additional manual checks (I am happy to see that aspect is happy :) ).

**Comment by tjhei** [comment]
Alright. Menno, let me know once this PR is ready.

**Comment by gassmoeller** [comment]
Just let me know when this is ready for review

**Comment by MFraters** [comment]
This pull request is now ready for review. Feel free to review to any depth you feel is necessary. Last year we agreed that the world builder is considered an external dependency so the level of scrutiny can be less, but as said, go as deep as you want to, I am always happy to get reviews :)

If you think the pull request if fine, please approve, but do not merge yet. Once approved for inclusion in Aspect, I will make the release official and this pull request can be merged :)

**Comment by MFraters** [comment]
I also wanted to mention that the pull request for the release is standing ready here: https://github.com/GeodynamicWorldBuilder/WorldBuilder/pull/275. You can see the coverage, some benchmarks and all the test environments I am testing the world builder for (and which pass), which are Ubuntu 20.04, 18.04 and 16.04, MacOs latest and Windows latest (both Visual studio and MinGW compilers). I also test with different (older) cmake versions down to cmake 2.8.10.

File: contrib/world_builder/CHANGELOG.md
Comment (tjhei) [inline contrib/world_builder/CHANGELOG.md:18]:
```suggestion
- Added an option for Continuous Monotone Spline interpolation. This is useful for faults and slabs which need to be continuous and smooth. \[Menno Fraters; 2021-05-24; [#130](github.com/GeodynamicWorldBuilder/WorldBuilder/issues/130),[#237](github.com/GeodynamicWorldBuilder/WorldBuilder/pull/237)\]
```

**Review by tjhei — APPROVED** [review]
Looks good to me.

**Comment by MFraters** [comment]
Thanks for the review Timo! By searching for `contious` in the code, I actually found more cases of this spelling mistake (all in documentation stuff). I am fixing them now :)

**Review by gassmoeller — APPROVED** [review]

**Pull request merged**
